### PR TITLE
Minor cleanup to CI tests

### PR DESCRIPTION
**Pull request by timburks**
- replace `make test` with `go test ./...`
- use built-in server for coverage testing

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -78,6 +78,21 @@ jobs:
     - name: Test registry server with PostgreSQL
       run: go test ./server/registry -postgresql
 
+    - name: Compute code coverage
```
Comment (timburks) [inline .github/workflows/go.yml:81]: I moved this earlier to clarify that it's *not* using the standalone server (notice that the `APG_REGISTRY` environment variable settings are also removed)

File: .github/workflows/go.yml [hunk 2]
```diff
@@ -78,6 +78,21 @@ jobs:
     - name: Test registry server with PostgreSQL
       run: go test ./server/registry -postgresql
```
Comment (timburks) [inline .github/workflows/go.yml:79]: Should we be clearing the test cache here?

Comment (theganyo) [reply .github/workflows/go.yml:79]: It won't do anything as `go test` hasn't been run before.

Comment (timburks) [reply .github/workflows/go.yml:79]: go test is being run on line 65 and again on line 79. Should we clear the cache before the run on line 79?

File: .github/workflows/go.yml (hunk 1 above)
Comment (theganyo) [reply .github/workflows/go.yml:81]: Is there any reason to not to combine this with the `Test everything with SQLite` test?

File: .github/workflows/go.yml (hunk 2 above)
Comment (timburks) [reply .github/workflows/go.yml:79]: From the runtime of the test, it appears to be doing something but there are no code changes. I wonder if changing the test flags is enough to clear the cache. *off to read docs*

Comment (timburks) [reply .github/workflows/go.yml:79]: "To disable test caching, use any test flag or argument other than the cacheable flags" https://pkg.go.dev/cmd/go/internal/test

Comment (theganyo) [reply .github/workflows/go.yml:79]: JFYI, a quick trick to bypass the cache is to add a count flag, ie. `go test ./... -count=1`.

File: .github/workflows/go.yml (hunk 1 above)
Comment (theganyo) [reply .github/workflows/go.yml:81]: Hmm. If we have postgres-specific code (where is that?) and that is our targeted production platform, then I think we should emit the coverage with the postgres test.

**Review by theganyo — APPROVED** [review]

File: .github/workflows/go.yml [hunk 3]
```diff
@@ -83,7 +83,7 @@ jobs:
         CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
       run: |
         go clean -testcache
-        go test -race -coverprofile=coverage.txt -covermode=atomic ./...
+        go test -race -coverprofile=coverage.txt -covermode=atomic -coverpkg=./server/... ./...
```
[the remark this answers is not in the document]
Comment (timburks) [reply .github/workflows/go.yml]: weirdly, when I changed to `coverpkg=./...`, the coverage dropped back to 50%.

[the remark this answers is not in the document]
Comment (theganyo) [reply .github/workflows/go.yml]: That's what I had seen as well.

[the remark this answers is not in the document]
Comment (timburks) [reply .github/workflows/go.yml]: I'm reverting any attempt to tweak `coverpkg` for this PR. Feel free to pick it up again.

**Pull request merged**